blog/_posts/2021-10-21-scala-3.1.0-released.md

```diff
@@ -0,0 +1,185 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.0 released!
+---
+Hello from the Scala 3 team! It has already been six weeks since we have announced the release candidate for the first minor version after the initial release of Scala 3. Now, after two more RCs, we can confidently promote Scala 3.1.0-RC3 to a new stable release of the language. In other words, Scala 3.1.0 is officially out!
+
+## Compatibility notice
+
+This is a first *minor* release after the initial release of Scala 3.0. This has following consequences:
+
+- Scala 3.1 is backward binary compatible: you can use dependencies compiled with Scala 3.0 in 3.1 projects.
+- Scala 3.1 is _not_ forward binary compatible: you _cannot_ use dependencies compiled with Scala 3.1 in Scala 3.0 projects.
+- Updating Scala to 3.1 is not a forward compatible change. If you are a library maintainer and are following SemVer you may want to update in your next minor release.
+
+Although we cannot guarantee full source compatibility between minor versions, we have put a lot of effort into assuring that all code that was working in 3.0.2, except some rare cases, will also work in 3.1.0. This means that if you are an application developer, you can confidently update the compiler version to take advantage of the newest improvements.
+
+If you are a library maintainer, *updating to 3.1.0 will force all of your users to need to update as well*. You may consider still publishing your library using 3.0.2 to ensure that users who are reluctant to update a compiler can still use it. On the other hand, we still encourage you to test your library using 3.1.0. We understand that the current state of binary compatibility may be unsatisfactory for library maintainers. We are actively working on improvements in this area in future releases of Scala. We are taking the problem of forward-compatibility seriously, and we will provide a technical solution for 3.2.0
```

Comment:
```suggestion
If you are a library maintainer, *updating to 3.1.0 will force all of your users to update to 3.1.0 as well*. We understand that the current state of binary compatibility may be unsatisfactory. We are actively working on technical solutions to support forward-compatibility in 3.2.0. In the meantime, we recommend testing your library with Scala 3.1.0 and 3.0.2, but _publishing_ it with 3.0.2. This will allow downstream users to use your library even if they can’t update to Scala 3.1.0. You can find an example of build that follows this recommendation [here](https://github.com/typelevel/scalacheck/pull/847).
```

(not 100% sure… I think it’s nice to provide an example that people can replicate, though)